components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -1786,12 +1468,556 @@ subroutine ma_convproc_tend(                                           &
    return
 end subroutine ma_convproc_tend
 
+!====================================================================================
+   subroutine compute_updraft_mixing_ratio(                             &
+                doconvproc_extd, icol,  ktop,   kbot,   iconvtype,      & ! in
+                dt,     mbsth,  dp_i,   dpdry_i,        cldfrac,        & ! in
+                rhoair_i,       zmagl,  dz,     mu_i,   eudp,           & ! in
+                const,  t,      aqfrac, icwmr,          rprd,           & ! in
+                fa_u,   dconudt_wetdep,         dconudt_activa,         & ! out
+                conu                                                    ) ! inout 
+!-----------------------------------------------------------------------
+! Compute updraft mixing ratios from cloudbase to cloudtop
+! No special treatment is needed at k=pver because arrays are dimensioned 1:pver+1
+! A time-split approach is used.  First, entrainment is applied to produce
+!    an initial conu(m,k) from conu(m,k+1).  Next, chemistry/physics are
+!    applied to the initial conu(m,k) to produce a final conu(m,k).
+!    Detrainment from the updraft uses this final conu(m,k).
+! Note that different time-split approaches would give somewhat different results
+!-----------------------------------------------------------------------
+   use ppgrid, only: pver, pverp, pcols
+   use constituents, only: pcnst
+
+   ! cloudborne aerosol, so the arrays are dimensioned with pcnst_extd = pcnst*2
+   integer,  parameter  :: pcnst_extd = pcnst*2
+   logical, intent(in)  :: doconvproc_extd(pcnst_extd) ! flag for doing convective transport
+   integer, intent(in)  :: icol                 ! column index
+   integer, intent(in)  :: ktop                 ! top level index
+   integer, intent(in)  :: kbot                 ! bottom level index
+   integer, intent(in)  :: iconvtype            ! 1=deep, 2=uw shallow
+   real(r8),intent(in)  :: dt                   ! Model timestep [s]
+   real(r8),intent(in)  :: mbsth                ! threshold below which we treat the mass fluxes as zero [mb/s]
+   real(r8),intent(in)  :: dp_i(pver)           ! dp [mb]
+   real(r8),intent(in)  :: dpdry_i(pver)        ! dp [mb]
+   real(r8),intent(in)  :: cldfrac(pcols, pver) ! cloud fraction [fraction]
+   real(r8),intent(in)  :: rhoair_i(pver)       ! air density at current i [kg/m3]
+   real(r8),intent(in)  :: zmagl(pver)          ! height above surface [m]
+   real(r8),intent(in)  :: dz                   ! layer thickness [m]
+   real(r8),intent(in)  :: mu_i(pverp)          ! mu at current i (note pverp dimension, see ma_convproc_tend) [mb/s]
+   real(r8),intent(in)  :: eudp(pver)           ! eu(i,k)*dp(i,k) at current i [mb/s]
+   real(r8),intent(in)  :: const(pcnst_extd,pver)   ! gathered tracer array [kg/kg]
+
+   real(r8), intent(in) :: t(pcols,pver)        ! Temperature [K]
```

Comment:
You will probably revisit this, so, please change these one-letter variables the next time.

Reply:
will revisit in the next PR